### PR TITLE
Add PollTimeout() to ensure a timeout actually occurs by using std::chrono

### PR DESCRIPTION
**Pull request by john-gress**

### CODE REVIEW

**Comment by KjellKod** [comment]
:+1:

**Comment by shwetabhandare** [comment]
Nice unit tests! Good job! 
:+1:

File: test/HarpoonKrakenTests.h [hunk 1]
```diff
@@ -1,22 +1,26 @@
 #pragma once
 
 #include <gtest/gtest.h>
+#include <FileIO.h>
 
 class HarpoonKrakenTests : public ::testing::Test {
 public:
 
-   HarpoonKrakenTests() {
+   HarpoonKrakenTests() : mIpc("ipc:///tmp/HarpoonKrakenTests.ipc"){
```
Comment (john-gress) [inline test/HarpoonKrakenTests.h]: Remove mIpc

**Comment by KjellKod** [comment]
:+1:

**Pull request merged**